include/zephyr/linker/devicetree_regions.h

```diff
@@ -187,6 +205,29 @@
 #define _CHECK_APPLY_FN(compat, EXPAND_MPU_FN, ...)					\
 	DT_FOREACH_STATUS_OKAY_VARGS(compat, _CHECK_ATTR_FN, EXPAND_MPU_FN, __VA_ARGS__)
 
+/**
+ * Check that the node_id has both properties:
+ *  - zephyr,memory-region-mmu
+ *  - zephyr,memory-region
+ *  - reg
+ *
+ * and call the EXPAND_MMU_FN() macro
+ */
+#define _CHECK_MMU_ATTR_FN(node_id, EXPAND_MMU_FN, ...)					\
+	COND_CODE_1(UTIL_AND_AND(DT_NODE_HAS_PROP(node_id, zephyr_memory_region_mmu),	\
```

Comment:
Is it really worth adding UTIL_AND_AND just to use it here? Not a blocking comment since I don't maintain the util header, but it doesn't seem justified since you can just nest COND_CODE_1 here. I think it is risky to start adding macros like UTIL_AND_AND -- where do you stop? UTIL_AND_OR? UTIL_OR_AND? UTIL_OR_OR? UTIL_AND_AND_AND? ...

Reply:
Okay, will remove this macro.